chain/src/chain.rs

```diff
@@ -524,8 +524,16 @@ impl BlockChain {
 
         let (state_root, multi_state) = {
             // if no txns, state_root is kept unchanged after calling txn-execution
-            let state_root1 = executed_data.state_root;
+            let state_root1 =
+                if starcoin_data_migration::should_do_migration(header.number(), header.chain_id())
+                {
+                    // Apply migration data if this is a migration block
```

Comment:
Find a way to isolate special procedure.

Reply:
fixed